### PR TITLE
Validate non-ascii characters 

### LINKED ISSUE #130: Warning non ascii recipes
As you know, Python 2 loads the python files as ascii by default. As we removed the shebang from our recipes, we need to check manually or wait for someone open an issue complaining about the recipe support.

@madebr has a good idea, add a hook which tries to load the conanfile with ascii encoding. As Python 2 is not a requirement, we can show a warning message instead of an error.

### PR DESCRIPTION
**Pull request by uilianries**
As Python 2 is close to its retirement, so I preferred keep this hook as an error. After 2020 we can change it to warning instead.

Related issue: https://github.com/conan-io/conan-center-index/pull/278#discussion_r343837509

/cc @madebr @SSE4 

closes #130

Wiki:

#### **<a name="KB-H038">#KB-H038</a>: "ASCII SUPPORT"**

Characters non-ascii are not compatible with Python 2, so they should be avoided.

### CODE REVIEW

File: tests/test_hooks/conan-center/test_conan-center.py [hunk 1]
```diff
@@ -462,3 +463,17 @@ def configure(self):
         output = self.conan(['create', '.', 'name/version@user/test'])
         self.assertIn('ERROR: [NO AUTHOR (KB-H037)] Conanfile should not contain author. '
                       'Remove \'author = (\'foo\', \'bar\')', output)
+
+    def test_invalid_ascii_recipe(self):
+        conanfile = textwrap.dedent("""\
+        from conans import ConanFile
+        class AConan(ConanFile):
+            description = "open-source file compression that uses the Burrows–Wheeler algorithm."
```
Comment (madebr) [inline tests/test_hooks/conan-center/test_conan-center.py]: By including this character, the `test_conan-center.py` source itself becomes non-ascii.
Maybe create the conanfile source as follows?
```
conanfile = b"""

        from conans import ConanFile
        class AConan(ConanFile):
            description = "open-source file compression that uses the Burrows\xe2\x80\x93Wheeler algorithm."
            def configure(self):
                pas
"""
```

Comment (uilianries) [reply tests/test_hooks/conan-center/test_conan-center.py]: hahaha! It's true! I gonna do it.

Comment (uilianries) [reply tests/test_hooks/conan-center/test_conan-center.py]: this was funny! it's fixed now!

**Comment by madebr** [comment]
I think this hook is ok in its current state.
Can this be merged?

**Comment by danimtb** [comment]
let's update it and merge it

**Comment by uilianries** [comment]
@danimtb All green again!

**Comment by danimtb** [comment]
I totally missed this one and I think it is a duplicate of #199

**Comment by uilianries** [comment]
And I created both PR :facepalm:

**Pull request closed without merging**